patch-upstream.sh

```diff
@@ -10,6 +10,18 @@ their Lambdas with Lambda Layers configured to export to the X-Ray backend.
 
 END_DOCUMENTATION
 
+# patch otel version on collector
+cd opentelemetry-lambda/collector
+
+PATCH_OTEL_VERSION="../../OTEL_Version.patch"
+
+if [ -f $PATCH_OTEL_VERSION ]; then
+    patch -p2 < $PATCH_OTEL_VERSION;
+fi
+
+# return to root directory after patch
+cd ../..
+
```

Comment:
I just realized that it would be better if to patch at the start, and then continue to[ patch/replace the directories](https://github.com/aws-observability/aws-otel-lambda/blob/main/patch-upstream.sh#L17). Cause patching after replacing with adot components seems confusing, as there are multiples files in [collector/](https://github.com/open-telemetry/opentelemetry-lambda/tree/f27bde282660babba9d3e468a841755a4c5dd6e7/collector) that needs to be patched

Reply:
Can I just make a request to not change what isn't broken? This has worked before and should continue to work. Unless there is measurable and strong reason to make this change I would prefer to leave it as is.

Reply:
Sure, its just bit tricky, i had to update the patch file to remove the lambda components part as it will be replaced anyways before the patch. Updated the PR